index.js

```diff
@@ -565,6 +565,9 @@ function Argv (processArgs, cwd) {
 
   function setPlaceholderKeys (argv) {
     Object.keys(options.key).forEach(function (key) {
+      // don't set placeholder keys for dot
+      // notation options 'foo.bar'.
+      if (key.match(/.*\..*/)) return
```

Comment:
Showing my ignorance, but is it necessary to use RegExp here? Could we not get away with `indexOf('.')` instead? Not sure [which](http://jsperf.com/test-vs-indexof-fast) [performs](http://jsperf.com/regexp-vs-indexof) [better](http://jsperf.com/indexof-vs-looping-charat), but I'd prefer the simplest possible solution.